spec/models/miq_report_result/result_set_operations_spec.rb

```diff
@@ -0,0 +1,75 @@
+describe MiqReportResult::ResultSetOperations do
+  describe ".result_set_for_reporting" do
+    let(:result_set_options_base) { {'limit' => '1000'} }
+
+    let(:user) { FactoryBot.create(:user_with_group) }
+
+    let(:result_set) do
+      [{"name" => "TEST_VM2", "size" => 115_533, "type" => 'small'},
+       {"name" => "TEST_VM1", "size" => 332_233, "type" => 'large'},
+       {"name" => "PROD_VM1", "size" => 112_233, "type" => 'large'}]
+    end
+
+    let(:report_result) { FactoryBot.create(:miq_report_result, :miq_group => user.current_group) }
+    let(:columns) { %w[name size type] }
+
+    let(:col_formats) { Array.new(columns.count) }
+
+    let!(:report) do
+      FactoryBot.create(:miq_report, :miq_group => user.current_group, :miq_report_results => [report_result],
+                        :col_order => columns, :col_formats => col_formats)
+    end
+
+    before do
+      allow(report_result).to receive(:result_set).and_return(result_set)
+      User.current_user = user
+    end
+
+    subject { report_result.result_set_for_reporting(result_set_options) }
+
+    let(:expected_result_set) do
+      [{"name" => "TEST_VM2", "size" => "112.8 KB", "type" => 'small'},
+       {"name" => "TEST_VM1", "size" => "324.4 KB", "type" => 'large'}]
+    end
+
+    let(:result_set_options) { result_set_options_base.merge(:filter_column => 'name', :filter_string => 'TEST') }
```

Comment:
verify: this line is basically the reason for this PR?

Reply:
probably not sure what do you asking for :) but filtering on this line has parameters of already existing feature (filter according to one parameter `:filter_column`).

this PR extends it to have possibility filter it according more columns - `:filter_column_1`, `:filter_column_2`,... and it is online 47, 56, ...